api/v1alpha1/types_etcd.go

```diff
@@ -208,6 +208,9 @@ type EtcdConfig struct {
 	// Quota defines the etcd DB quota.
 	// +optional
 	Quota *resource.Quantity `json:"quota,omitempty"`
+	// SnapshotCount defines the number of committed transactions to trigger a snapshot to disk
+	// +optional
+	SnapshotCount *int `json:"snapshotCount,omitempty"`
```

Comment:
1. once we include this as a configurable parameter, we will also have to check what happens when this parameter is updated since this is now going to be part of an etcd spec.
2. The description can simply be: `configures the number of applied Raft entries to hold in-memory before compaction` and also provide a link to https://etcd.io/docs/v3.4/op-guide/maintenance/#raft-log-retention

Reply:
> we will also have to check what happens when this parameter is updated since this is now going to be part of an etcd spec.

Sorry, I didn't get this. If the param is updated, then in the next reconciliation it will be reflected in the configMap. Is your question about how etcd container reacts to this change? I've tried this locally and it didn't affect the cluster in anyway. Pls correct me if this is about something else. 

> The description can simply be: configures the number of applied Raft entries to hold in-memory before compaction

Sure, this is more readable. I'll update.